### PR TITLE
chore: add snippets for dev docs, corrected some docstring examples in cache client

### PR DESCRIPTION
**Pull request by anitarua**
Addresses https://github.com/momentohq/client-sdk-rust/issues/210

### CODE REVIEW

File: example/Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,5 @@ edition = "2021"
 #momento = "0.15.0"
 momento = { path = "../" }
```
Comment (anitarua) [inline example/Cargo.toml:12]: examples are currently using the local SDK, but once we cut a release we can update this here

**Review by malandis — COMMENTED** [review]
Thanks for doing this. As a discussion point, I see two things:
1. In the examples should we use the ?-operator to propagate errors up from the client calls?
2. Should we test for particular error cases in some examples?

For example in some examples we use the ?-operator on the cache calls, then display the answer in the happy case.

On others we match on the result and display an error.

On others we test for particular error codes, eg:

```rust
    match cache_client.set(cache_name, "key", "value").await {
        Ok(_) => println!("Set successful"),
        Err(e) => {
            if let MomentoErrorCode::NotFoundError = e.error_code {
                println!("Cache not found: {}", cache_name);
            } else {
                eprintln!("Error setting value in cache {}: {}", cache_name, e);
            }
        }
    }
```

My inclination given Rust's features is to:
1. Bubble up the errors using `?`
2. Match on all the other cases (hit vs miss, stored vs not stored, cache created vs already exists)

We can socialize the options with the team during the bug bash.

File: example/src/bin/docs_examples.rs [hunk 2]
```diff
@@ -157,16 +123,8 @@ pub async fn example_API_Increment(
...
-        Err(e) => {
-            if let MomentoErrorCode::NotFoundError = e.error_code {
-                println!("Cache not found: {}", cache_name);
-            } else {
-                eprintln!("Error incrementing value in cache {}: {}", cache_name, e);
-            }
-        }
-    }
+    cache_client.increment(cache_name, "key", 1).await?;
+    println!("Value incremented");
```
Comment (malandis) [inline example/src/bin/docs_examples.rs:127]: For this one the response has the new value, which is valuable to print. Can you update the print statement for that?

File: example/src/bin/docs_examples.rs [hunk 3]
```diff
@@ -468,28 +372,50 @@ pub async fn example_API_ListRemoveValue(
...
-pub async fn example_API_DictionarySetFields(
+pub async fn example_API_DictionaryIncrement(
     cache_client: &CacheClient,
     cache_name: &String,
 ) -> Result<(), MomentoError> {
-    match cache_client
+    cache_client
+        .dictionary_increment(cache_name, "dictionary_name", "field", 1)
+        .await?;
+    println!("Incremented field in dictionary");
```
Comment (malandis) [inline example/src/bin/docs_examples.rs:390]: Same comment here re: incremented value in response.

**Review by malandis — COMMENTED** [review]
Couple small issues. Looks great otherwise and much improved and cleaner.

File: example/src/bin/docs_examples.rs (hunk 3 above)
Comment (anitarua) [reply example/src/bin/docs_examples.rs:390]: oh good catch, will update

**Review by malandis — APPROVED** [review]

**Pull request merged**